lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestContextStack.cs

```diff
@@ -0,0 +1,226 @@
+        public void TestContextStackingOnLocalRuntime()
+        {
+            string testFolder = DefaultRuntimeFolder + Guid.NewGuid().ToString("N").Substring(0, 4);
+            CleanUp(testFolder);
+            TestRun(DriverConfigurations(), typeof(ContextStackHandlers), 1, "testContextStack", "local", testFolder);
+            ValidateSuccessForLocalRuntime(1, testFolder: testFolder);
+            ValidateMessageSuccessfullyLogged(TaskValidationMessage, testFolder);
+            ValidateMessageSuccessfullyLogged(ClosedContextValidationMessage, testFolder);
+            CleanUp(testFolder);
+        }
+
+        public IConfiguration DriverConfigurations()
+        {
+            var helloDriverConfiguration = DriverConfiguration.ConfigurationModule
+                .Set(DriverConfiguration.OnDriverStarted, GenericType<ContextStackHandlers>.Class)
+                .Set(DriverConfiguration.OnEvaluatorAllocated, GenericType<ContextStackHandlers>.Class)
+                .Set(DriverConfiguration.OnContextActive, GenericType<ContextStackHandlers>.Class)
+                .Set(DriverConfiguration.OnTaskMessage, GenericType<HelloTaskMessageHandler>.Class)
+                .Set(DriverConfiguration.OnTaskCompleted, GenericType<ContextStackHandlers>.Class)
+                .Set(DriverConfiguration.OnTaskRunning, GenericType<ContextStackHandlers>.Class)
+                .Set(DriverConfiguration.OnEvaluatorCompleted, GenericType<ContextStackHandlers>.Class)
+                .Set(DriverConfiguration.OnContextClosed, GenericType<ContextStackHandlers>.Class)
+                .Build();
+
+            return TangFactory.GetTang().NewConfigurationBuilder(helloDriverConfiguration)
+                .BindSetEntry<DriverBridgeConfigurationOptions.SetOfAssemblies, string>(typeof(HelloTask).Assembly.GetName().Name)
+                .BindSetEntry<DriverBridgeConfigurationOptions.SetOfAssemblies, string>(typeof(TestContextStack).Assembly.GetName().Name)
+                .BindNamedParameter<IsRetain, bool>(GenericType<IsRetain>.Class, "false")
+                .BindNamedParameter<NumberOfEvaluators, int>(GenericType<NumberOfEvaluators>.Class, "1")
+                .Build();
+        }
+
+        private sealed class ContextStackHandlers : 
+            IObserver<IDriverStarted>,
+            IObserver<IAllocatedEvaluator>,
+            IObserver<IActiveContext>,
+            IObserver<ICompletedTask>,
+            IObserver<IRunningTask>,
+            IObserver<IFailedTask>,
+            IObserver<ICompletedEvaluator>,
+            IObserver<IClosedContext>
+        {
+            private readonly IEvaluatorRequestor _requestor;
+
+            [Inject]
+            private ContextStackHandlers(IEvaluatorRequestor evaluatorRequestor)
+            {
+                _requestor = evaluatorRequestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().Build());
```

Comment:
This relies on defaults for the number of Evaluators and their resource footprint. While they are unlikely to change, maybe it is better to call these out explicitly here?